src/shared/forms/controllers/PasswordInputController.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { useForm, ErrorOption } from 'react-hook-form'
+
+import { PasswordInputController } from 'shared/forms/controllers/PasswordInputController'
+import { fireEvent, render, screen } from 'tests/utils'
+
+type PasswordForm = {
+  password: string
+}
+
+describe('<PasswordInputController />', () => {
+  it('should not show error when form is invalid and password is empty', async () => {
+    renderPasswordInputController({
+      error: { type: 'custom', message: 'error' },
+    })
+
+    expect(screen.queryByText('error')).toBeNull()
+  })
+
+  it('should show error when form is invalid and password is not empty', async () => {
+    renderPasswordInputController({
+      error: { type: 'custom', message: 'error' },
+    })
+
+    const input = screen.getByPlaceholderText('Ton mot de passe')
+    fireEvent.changeText(input, 'pass')
+
+    expect(screen.getByText('error')).toBeTruthy()
+  })
+
+  it('should show that password is mandatory', async () => {
+    renderPasswordInputController({})
+
+    expect(screen.getByText('Obligatoire')).toBeTruthy()
+  })
+
+  it.each([
+    '12 Caractères',
+    '1 Majuscule',
```

Comment:
Pour moi il y a une amélioration possible ici mais je suis peut-être méga $@&%

Tu as `<PasswordRule title="1 Majuscule" isValidated={containsCapital(password)} />` dans PasswordSecurityRules

Et ici tu re-écris "1 Majuscule"

Maintenant si on change le détail d'une règle le test est pété...

Alors qu'il pourrait y avoir un tableau [..., {name:'A Majuscule', rule:containsCapital}] et
`<PasswordRule title="{arrayOfRules[index].name}" isValidated={arrayOfRules[index].rule(password)} />`

et ton `it.each(arrayOfRules.keys())` ou un truc du genre

Do you see what i mean ?

Reply:
je pense que c'est un refactoring intéressant mais à faire dans une autre PR